### ISSUE TITLE
Configuring SSL For DefectDojo

### ISSUE THREAD

**Issue by ranjithprethan** [DefectDojo/django-DefectDojo#2534]
Hi,

I am running DD on Docker and trying to configure SSL. It seems the SSL documentation is not very clear to me. As per the installation steps given in the document under " Run Docker with HTTPS- Use your own Credentials" , 

- Generate a private key without password . Keytool can't generate a JKS ( keystore file ) without a password. so can we add an password attribute in the configuration file. 
- Edit the conf file . ( I believe we need to edit the nginx_TLS.conf file , but it's not mentioned in the document)
-Run **rm -f docker-compose.override.yml** and **ln -s docker-compose.override.https.yml docker-compose.override.yml** ( Do we need to run both?)

Can the SSL documentation be revisited please?

https://github.com/DefectDojo/django-DefectDojo/blob/master/DOCKER.md

Thanks
Ranjith

**Comment by valentijnscholten** [comment]
The docs look OK to me. You mention Keytool and jks files. These are of no use here. You need the keypair in crt/pem format, as with any apache or nginx ssl setup. We don't have the resources to document every single possible setup.

**Comment by ranjithprethan** [comment]
Hi,

Thank you for your support.
The documentation is really confusing. Let me explain it step by step.

1. I created the private key and CSR using openssl, and created ssl cert .crt from my PKI
2. I moved my key and crt files into nginx folder 

 /root/django-DefectDojo/nginx/ddtest.com.crt
 /root/django-DefectDojo/nginx/ddtest.key

3. I renamed nginx_TLS.conf as nginx.conf and the configuration is given below

 server {
    server_tokens off;
    listen 8443 ssl;
    server_name server123.com;
    ssl_certificate /root/django-DefectDojo/nginx/ddtest.com.crt;
    ssl_certificate_key /root/django-DefectDojo/nginx/ddtest.key;

4. I used yaml validator to validate yaml file, here is how my docker-compose.override.https.yaml looks like 

services: 
  nginx: 
    environment: 
      GENERATE_TLS_CERTIFICATE!: "true"
      USE_TLS: "true"
    ports: 
      - 
        mode: host
        protocol: tcp
        published: "${DD_PORT:-8443}"
        target: "${DD_PORT:-8443}"
  uwsgi: 
    environment: 
      DD_CSRF_COOKIE_SECURE: "True"
      DD_SESSION_COOKIE_SECURE: "True"
version: "3.7"

Finally when i restarted the docker compose i get this error below,

### 2020/06/15 05:06:43 [emerg] 1#1: cannot load certificate "/etc/nginx/ssl/nginx.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/ssl/nginx.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
nginx_1         | nginx: [emerg] cannot load certificate "/etc/nginx/ssl/nginx.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/ssl/nginx.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)
django-defectdojo_nginx_1 exited with code 1

Any help is much appreciated.

Thanks

**Comment by JPMartinezz** [comment]
I'm getting the same thing (with same config). Based on the error, it looks like the crt is not being pulled from the DefectDojo/nginx folder. Any help would be greatly appreciated.

**Comment by valentijnscholten** [comment]
what happens if you follow the instructions linked to above?

**Comment by JPMartinezz** [comment]
I get the same error **"cannot load certificate "/etc/nginx/ssl/nginx.crt": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/nginx/ssl/nginx.crt','r') error:2006D080:BIO routines:BIO_new_file:no such file)"**

**Comment by JPMartinezz** [comment]
@ranjithprethan - i got SSL configured after modifying the Dockerfile.nginx file to include additional steps:

The below to manually copy the key and cert:
COPY nginx/nginx.key /etc/nginx/ssl/nginx.key
COPY nginx/nginx.crt /etc/nginx/ssl/nginx.crt

The below to change the permissions on the cert and key (after they've been copied over):
RUN \
  chmod (whatever permissions config youd like here) /etc/nginx/ssl/nginx.key && \
  true

**Comment by ranjithprethan** [comment]
Hi JP

I get the same error. would it be possible to provide all the steps that you taken to configure SSL?

- Changes to - nginx dir ( nginx.conf and nginx_TLS.conf)
- Changes to docker-compose.override.https.yml
- symbolic link (docker-compose.override.yml -> docker-compose.override.https.yml) Infact there is no docker-compose.override.yml file available in the project file
Changes to Dockerfile.nginx file
![SSL Config](https://user-images.githubusercontent.com/26608216/87092693-e32b9580-c209-11ea-8ae7-00edd6138b12.JPG)



Thanks

**Comment by JPMartinezz** [comment]
@ranjithprethan - I removed the symbolic link step because (as you mentioned) there was no docker-compose.override.yml file to begin with and I was encountering more issues when performing that step. I recommend just running with the provided docker-compose.yml file. 

I noticed you are running the COPY commands prior to the directory being created and modified (in the RUN step). I included mine after like shown below

![image](https://user-images.githubusercontent.com/39783366/87229915-061c8d80-c361-11ea-8bf0-4264a65d5419.png)

I then set the nginx.conf ssl_certificate & ssl_certificate_key paths like below:

![image](https://user-images.githubusercontent.com/39783366/87229945-4aa82900-c361-11ea-9628-9c866ab9581c.png)


Hope this helps!
